src/utils.rs

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+extern crate serde_json;
+
+// Macros to help with json serializing of undeclared structs.
+// json_value!({ }) retunns a serde_json::value::Value from an anonymous struct.
+// json!({ }) returns a string from an anonymous struct.
+// Combining both macros let you build arbitrary complex json objects easily.
+//
+// Examples: json!({ type: "light", value: 1.2, on: true });
+//           json!({ type: "notification", value: json_value!({ r: 54, g: 12, b: 88 }) });
+
+#[macro_export]
+macro_rules! json {
+    ({ $($i:ident: $v:expr),* }) => {
+        {
+            use std::collections::BTreeMap;
+            let mut map: BTreeMap<String, serde_json::Value> = BTreeMap::new();
+            $(
+                map.insert(String::from(stringify!($i)), serde_json::to_value(&$v));
+            )*
+            serde_json::to_string(&map).unwrap_or("{}".to_owned())
+        }
+    }
+}
+
+#[macro_export]
+macro_rules! json_value {
+    ({ $($i:ident: $v:expr),* }) => {
```

Comment:
Same in this macro. Looks like it may sufficient to add another branch for *([...])* and then use *Vec<...>* inside.